### PR TITLE
Refactor how config is handled

### PR DESCRIPTION
**Pull request by astrofrog**
I have the following proposal: bot instances (e.g. astropy-bot or sunpy-bot) can define a 'global' ``pyproject.toml`` file that gets read in just after the app is instantiated. It is then stored in a configuration object at ``app.conf`` (``app.config`` is already used by flask). When a user requests the configuration for a given repo, we then copy the global config and update it with the config read from the repo. If we do this, we can then also get rid of the ``app.setting_name_default`` and simply edit ``app.conf`` directly if really needed (otherwise we just edit the toml file directly).

This global config can then also be read by parts of baldrick that don't depend on the app by directly reading the global settings file - this will make it easy to configure e.g. the stale issues/pull requests scripts at the level of e.g. astropy-bot rather than individual repositories.

@Cadair - what do you think? If you agree, I can try and work on this more (this PR is not at all finished)

One open question then is whether the configuration object should behave like a dict, or whether it should have a ``sections`` attribute for instance like the PR does currently, which itself is a dict.

### CODE REVIEW

File: baldrick/__init__.py [hunk 1]
```diff
@@ -31,6 +32,13 @@ def create_app(name, register_blueprints=True):
 
     app.wsgi_app = ProxyFix(app.wsgi_app)
 
+    # Check if there is a global configuration
+    global_toml = os.path.join(os.path.expanduser('~'), 'pyproject.toml')
```
Comment (Cadair) [inline baldrick/__init__.py]: why `~` and not just the current directory?

Comment (astrofrog) [reply baldrick/__init__.py]: Hmm yeah maybe that makes more sense.

File: baldrick/config.py [hunk 2]
```diff
@@ -9,20 +9,14 @@ def loads(text, tool='baldrick'):
     return Config(toml.loads(text)['tool'][tool])
 
 
-class Config:
+class Config(dict):
```
Comment (Cadair) [inline baldrick/config.py]: `UserDict`?

**Comment by Cadair** [comment]
lgtm bar the little comment. Feel free to go for it.

File: baldrick/config.py (hunk 2 above)
Comment (astrofrog) [reply baldrick/config.py]: It looks like UserDict is now obsolete - https://legacy.python.org/dev/peps/pep-3108/

Comment (astrofrog) [reply baldrick/config.py]: (in Python 3.x)

Comment (Cadair) [reply baldrick/config.py]: but its still here: https://docs.python.org/3/library/collections.html?highlight=collections#collections.UserDict :/

Comment (Cadair) [reply baldrick/config.py]: also 3108 says "Useful bits moved to the 'collections' module."

Comment (Cadair) [reply baldrick/config.py]: Having said that, I am fine with this the way it is.

**Pull request merged**